### DISCUSSION TITLE
Sandboxie plus protection against brutal malwares.

### DISCUSSION THREAD

**Discussion by catalcin** [sandboxie-plus/Sandboxie#2784] [Q&A Feedback]
Does sandboxie plus protects me from really dangerous viruses as "Rat's" "WannaCry" "NoEscape"?

I'm not running these with that intention but theres some sketchy programs promising stuff it doesnt seem like.

**Comment by bastik-1001** [comment]
There are no guarantees that Sandboxie will prevent every malware from escaping the sandbox. It also depends on the kind of malware. By default, some program running in a sandbox has read access to what the user has read access to. This may include access to passwords, cookies, tokens, or other types of credentials, which could be stolen. This will be true for RATs, but those will not be able to make changes to the system, unless they find a way out. Sandboxie wasn't originally designed with privacy in mind, as it seems. With newer versions, there is a privacy mode.

For ransomware, like WannaCry, Sandboxie will do better, as long as the malware is being contained within the sandbox, since any attempt to write to a file will create a copy within the sandbox, leaving the original file intact. One might run out of space, since all the files that are supposed to be overwritten with an encrypted version, will be copied and encrypted in the sandbox.

Newer versions of Sandboxie also have a security hardened box. The old hardened boxes, were 'only' disallowing some program to start with administrative rights, whereas the newer version restricts the availability to SysCalls.

As software will have flaws, there will be ways to escape sandboxes. There have been escapes from virtual machines and sandboxes.

Some malware tries to analyze its environment and if it is being run in virtualized (e.g. a VM or a sandbox) to alter its behavior to stay undetected by behavioral analysis. Some simple do nothing, others just terminate themselves.

I personally see Sandboxie as another line of defense, so if some website exploits my browser, it will be contained within the sandbox. I bet on the idea that there is not enough interest to try to escape the sandbox, since that is another step that is not required for most of the targets that can be attacked.

Someone being motivated to escape the sandbox, be it by being pressured due to circumstances, or academic interest, or simply for the fun of it, is most likely going to find a way to do that.
